app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeListFragment.kt

```diff
@@ -0,0 +1,131 @@
+
+        @JvmStatic
+        fun newInstance(sidePane: Boolean): ChangeListFragment {
+            return ChangeListFragment().withArguments(BundleKeys.SIDEPANE to sidePane)
+        }
+
+    }
+
+    private var onSessionListClickListener: OnSessionListClick? = null
+    private lateinit var changesList: List<Session>
+    private var sidePane = false
+
+    private lateinit var currentListView: ListView
+
+    /**
+     * The Adapter which will be used to populate the ListView/GridView with Views.
+     */
+    private lateinit var changeListAdapter: ChangeListAdapter
+
+    private val viewModelFactory by lazy { ChangeListViewModelFactory(appRepository) }
+    private val viewModel: ChangeListViewModel by viewModels { viewModelFactory }
+
+    @MainThread
+    @CallSuper
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val arguments = requireArguments()
+        sidePane = arguments.getBoolean(BundleKeys.SIDEPANE)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val contextThemeWrapper: Context = ContextThemeWrapper(requireContext(), R.style.Theme_AppCompat_Light)
+        val localInflater = inflater.cloneInContext(contextThemeWrapper)
+        val fragmentLayout = if (sidePane) R.layout.fragment_session_list_narrow else R.layout.fragment_session_list
+        val headerLayout = if (sidePane) R.layout.changes_header else R.layout.header_empty
+        val fragmentView = localInflater.inflate(fragmentLayout, container, false)
+        val headerView = localInflater.inflate(headerLayout, null, false)
+        currentListView = fragmentView.requireViewByIdCompat<ListView>(android.R.id.list)
+        currentListView.addHeaderView(headerView, null, false)
+        currentListView.setHeaderDividersEnabled(false)
+        return fragmentView
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeViewModel()
+    }
+
+    private fun observeViewModel() {
+        viewModel.changeListParameter.observe(viewLifecycleOwner) { (sessions, numDays, useDeviceTimeZone) ->
+            changesList = sessions
+            val adapter = ChangeListAdapter(requireContext(), sessions, numDays, useDeviceTimeZone)
+            currentListView.adapter = adapter
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        NotificationHelper(requireContext()).cancelScheduleUpdateNotification()
+        viewModel.updateScheduleChangesSeen(true)
+    }
+
+    @MainThread
+    @CallSuper
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        onSessionListClickListener = try {
+            context as OnSessionListClick
+        } catch (e: ClassCastException) {
+            throw ClassCastException("$context must implement OnSessionListClick")
+        }
```

Comment:
:x: I actually want to throw an exception. This check is here to fail fast if the Fragment/Activity setup has been done wrong.

Reply:
I think this still does not invalidates my suggestion, since you still can throw the exception without provoking that `context as OnSessionListClick` is throwing an additional one.

Reply:
:bulb: Now I understand. Is this what you suggest?

``` kotlin
if (context is OnSessionListClick) {
    onSessionListClickListener = context
} else {
    error("$context must implement OnSessionListClick")
}
```